packages/block-library/src/media-text/media-container.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import { IconButton, ResizableBox, Toolbar } from '@wordpress/components';
+import {
+	BlockControls,
+	MediaPlaceholder,
+	MediaUpload,
+} from '@wordpress/editor';
+
+/**
+ * Constants
+ */
+const ALLOWED_MEDIA_TYPES = [ 'image', 'video' ];
+
+class MediaContainer extends Component {
+	renderToolbarEditButton() {
+		const { mediaId, onSelectMedia } = this.props;
+		return (
+			<BlockControls>
+				<Toolbar>
+					<MediaUpload
+						onSelect={ onSelectMedia }
+						allowedTypes={ ALLOWED_MEDIA_TYPES }
+						value={ mediaId }
+						render={ ( { open } ) => (
+							<IconButton
+								className="components-toolbar__control"
+								label={ __( 'Edit Media' ) }
+								icon="edit"
+								onClick={ open }
+							/>
+						) }
+					/>
+				</Toolbar>
+			</BlockControls>
+		);
+	}
+
+	renderImage() {
+		const { mediaAlt, mediaUrl, className } = this.props;
+		return (
+			<Fragment>
+				{ this.renderToolbarEditButton() }
+				<figure className={ className }>
+					<img src={ mediaUrl } alt={ mediaAlt } />
+				</figure>
+			</Fragment>
+		);
+	}
+
+	renderVideo() {
+		const { mediaUrl, className } = this.props;
+		return (
+			<Fragment>
+				{ this.renderToolbarEditButton() }
+				<figure className={ className }>
+					<video controls src={ mediaUrl } />
+				</figure>
+			</Fragment>
+		);
+	}
+
+	renderPlaceholder() {
+		const { onSelectMedia, className } = this.props;
+		return (
+			<MediaPlaceholder
+				icon="format-image"
+				labels={ {
+					title: __( 'Media area' ),
+					name: __( 'a media file (image or video)' ),
+				} }
+				className={ className }
+				onSelect={ onSelectMedia }
+				accept="image/*,video/*"
+				allowedTypes={ ALLOWED_MEDIA_TYPES }
+			/>
+		);
+	}
+
+	render() {
+		const { mediaPosition, mediaUrl, mediaType, mediaWidth, commitWidthChange, onWidthChange } = this.props;
+		if ( mediaType && mediaUrl ) {
+			const onResize = ( event, direction, elt ) => {
+				onWidthChange( parseInt( elt.style.width ) );
+			};
+			const onResizeStop = ( event, direction, elt ) => {
+				commitWidthChange( parseInt( elt.style.width ) );
+			};
+			const enablePositions = {
+				right: mediaPosition === 'left',
+				left: mediaPosition === 'right',
+			};
+
+			let mediaElement = null;
+			switch ( mediaType ) {
+				case 'image':
+					mediaElement = this.renderImage();
+					break;
+				case 'video':
+					mediaElement = this.renderVideo();
+					break;
+			}
+			return (
+				<ResizableBox
+					className="editor-media-container__resizer"
```

Comment:
This doesn't adhere to class naming conventions.

https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#naming